design/pipeline-artifacts.md

```diff
@@ -47,9 +47,16 @@ Artifacts are downloaded either to `$PIPELINE_RESOURCESDIRECTORY` or to the dire
 
 ```yaml
 - download: string # identifier for the pipeline resource from which to download artifacts, optional; "current" means the current pipeline
-  artifact: string # identifier for the artifact to download; if left blank, downloads all artifacts associated with the resource provided
-  patterns: string | [ string ] # a minimatch path or list of [minimatch paths](tasks/file-matching-patterns.md) to download; if blank, the entire artifact is downloaded
+  artifact: string # identifier for the artifact to download; required
+  patterns: string # a minimatch path or list of [minimatch paths](tasks/file-matching-patterns.md) to download; if blank, the entire artifact is downloaded
   root: string # the directory in which to download files, defaults to $PIPELINE_RESOURCESDIRECTORY; if a relative path is provided, it will be rooted from $SYSTEM_DEFAULTWORKINGDIRECTORY
+  displayName: string # friendly name displayed in the UI
+  name: string # identifier for this step (A-Z, a-z, 0-9, and underscore)
+  condition: string
+  conditionOnError: boolean # 'true' if future steps should run even if this step fails; default to 'false'
+  enabled: true # whether or not to run this ste; defaults to 'true'
+  timeoutInMinutes: number
```

Comment:
These are core step properties. Can they all be set as well?

Note, this is also relevant to whether we change `artifact` to be required. If not, then the `timeout` property doesn't make sense when the single `download` step expands to multiple steps.

Reply:
I think line 56 should be "continueOnError". I'm in favor of adding these core properties, but it's low priority.